src/main/java/life/qbic/App.java

```diff
@@ -1,159 +1,76 @@
+  /**
+   * logs into OpenBIS asks for and verifies password
+   *
+   * @param commandLineParameters
+   * @return
+   */
+  private static QbicDataDownloader loginToOpenBIS(
+      PostmanCommandLineOptions commandLineParameters) {
+    System.out.format("Please provide password for user '%s':\n", commandLineParameters.user);
 
-                LOG.info(String.format("Number of files found: %s", foundSuffixFilteredIDs.size()));
+    String password = OpenBISPasswordParser.readPasswordFromInputStream();
 
-                downloadFilteredIDs(qbicDataLoader, ident, foundSuffixFilteredIDs);
-            }
-            // a regex pattern was provided -> only download files which contain the regex pattern
-        } else if (!commandLineParameters.regexPatterns.isEmpty()) {
-            for (String ident : commandLineParameters.ids) {
-                LOG.info(String.format("Downloading files for provided identifier %s", ident));
-                List<IDataSetFileId> foundRegexFilteredIDs = qbicDataLoader.findAllRegexFilteredIDs(ident, commandLineParameters.regexPatterns);
-
-                LOG.info(String.format("Number of files found: %s", foundRegexFilteredIDs.size()));
-
-                downloadFilteredIDs(qbicDataLoader, ident, foundRegexFilteredIDs);
-            }
-        } else {
-            // no suffix or regex was supplied -> download all datasets
-            for (String ident : commandLineParameters.ids) {
-                LOG.info(String.format("Downloading files for provided identifier %s", ident));
-                List<DataSet> foundDataSets = qbicDataLoader.findAllDatasetsRecursive(ident);
-
-                LOG.info(String.format("Number of data sets found: %s", foundDataSets.size()));
-
-                if (foundDataSets.size() > 0) {
-                    LOG.info("Initialize download ...");
-                    int datasetDownloadReturnCode = -1;
-                    try {
-                        datasetDownloadReturnCode = qbicDataLoader.downloadDataset(foundDataSets);
-                    } catch (NullPointerException e) {
-                        LOG.error("Datasets were found by the application server, but could not be found on the datastore server for "
-                                + ident + "." + " Try to supply the correct datastore server using a config file!");
-                        e.printStackTrace();
-                    }
-
-                    if (datasetDownloadReturnCode != 0) {
-                        LOG.error("Error while downloading dataset: " + ident);
-                    } else {
-                        LOG.info("Download successfully finished.");
-                    }
-
-                } else {
-                    LOG.info("Nothing to download.");
-                }
-            }
-        }
+    if (password.isEmpty()) {
+      System.out.println("You need to provide a password.");
+      System.exit(1);
```

Comment:
Maybe throw an exception you catch in the main method to avoid the system from terminating anywhere except the main method. It could lead to problems if the whole application is terminated from different locations in the code.